pkg/agent/openflow/pipeline.go

```diff
@@ -2106,22 +2106,32 @@ func (f *featureNetworkPolicy) conjunctiveMatchFlow(tableID uint8, matchPairs []
 }
 
 // defaultDropFlow generates the flow to drop packets if the match condition is matched.
-func (f *featureNetworkPolicy) defaultDropFlow(table binding.Table, matchPairs []matchPair) binding.Flow {
+func (f *featureNetworkPolicy) defaultDropFlow(table binding.Table, matchPairs []matchPair, enableLogging bool) binding.Flow {
 	cookieID := f.cookieAllocator.Request(f.category).Raw()
 	fb := table.BuildFlow(priorityNormal)
 	for _, eachMatchPair := range matchPairs {
 		fb = f.addFlowMatch(fb, eachMatchPair.matchKey, eachMatchPair.matchValue)
 	}
+	fb = fb.Action().Drop()
+
+	var customReason int
 	if f.enableDenyTracking {
-		return fb.Action().Drop().
-			Action().LoadRegMark(DispositionDropRegMark, CustomReasonDenyRegMark).
-			Action().SendToController(uint8(PacketInReasonNP)).
-			Cookie(cookieID).
-			Done()
+		customReason += CustomReasonDeny
+		fb = fb.
+			Action().LoadRegMark(DispositionDropRegMark, CustomReasonDenyRegMark)
```

Comment:
If you load `CustomReasonField` in line 2131, you don't need to load `CustomReasonDenyRegMark` here.

Reply:
Could `DispositionDropRegMark` be load inside `if enableLogging || f.enableDenyTracking {}`?

Reply:
Makes sense 💯